### DISCUSSION TITLE
Troubleshooting why cache for Server Components does not work

### DISCUSSION THREAD

**Discussion by pjanickovic-ph** [vercel/next.js#68808] [App Router]
Hello,

I do have a dashboard based on AppRouter 14 (latest version), data for dashboard are coming from a redis server, redis is updated every 5 minutes, and my goal is leverage the cache,

but from the logs i can clearly see, the server components code is being executed with every refresh of a site. I would like to investigate why, what is causing the opt out from cache. 

I am not using noStore(), headers(), cookies() in the Page components and downstream,
might it be because in the layout I use getSession from next-auth? 

can you please suggest how to investigate ? Thanks

**Comment by 1Mouse** [comment]
yeah it's because of the getSession 

alternatively use `useSession` in a `client component` this will cahce the page and partially hydrate the client component only on the browser. 
and u still got the benefits of server rendering of all other parts and caching.

**Comment by pjanickovic-ph** [comment]
Thanks for the quick response,

now It makes sense, why neither of the component get cached, every server
components is HOC wrapped, where session is used -> user level permissions

any thoughts ?


------------------------------
*From:* Mohamed Elfar ***@***.***>
*Sent:* Monday, August 12, 2024 22:01
*To:* vercel/next.js ***@***.***>
*Cc:* pjanickovic-ph ***@***.***>; Author <
***@***.***>
*Subject:* Re: [vercel/next.js] Troubleshooting why cache for Server
Components does not work (Discussion #68808)


yeah it's because of the getSession

alternatively use useSession in a client component this will cahce the page
and partially hydrate the client component only on the browser.
and u still got the benefits of server rendering of all other parts and
caching.

—
Reply to this email directly, view it on GitHub
<https://github.com/vercel/next.js/discussions/68808#discussioncomment-10318416>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/A3BDXBPND3AFI6QRDLLARZDZREIAXAVCNFSM6AAAAABMMZRQKGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZRHA2DCNQ>
.
You are receiving this because you authored the thread.Message ID:
***@***.***>
